docs/Middleware-Guide.md

```diff
@@ -0,0 +1,82 @@
+2. Next, the operator needs to register with the middleware on chain via a middleware-specific registry contract (see the *Guide To Provided Middleware Contracts* section for examples). To integrate with EigenLayer, the middleware's Registry contract provides a registration endpoint that calls on the `recordFirstStakeUpdate(..)` in the middleware's ServiceManager contract which in turn calls the `recordFirstStakeUpdate(..)` function in `Slasher.sol`. On successful execution of this function call, the event `MiddlewareTimesAdded(..)` is emitted and the operator has to start serving the tasks from the middleware.
+
+The following figure illustrates the above flow: 
+![Operator opting-in](./images/operator_opting.png)
+
+
+### *Deregistering from middleware*
+In order for any EigenLayer operator to be able to de-register from a middleware, EigenLayer provides the interface `recordLastStakeUpdateAndRevokeSlashingAbility(..)`. Essentially, in order for an operator to de-register from a middleware, the operator has to call `recordLastStakeUpdateAndRevokeSlashingAbility(..)` in EigenLayer's `Slasher.sol` via the middleware's ServiceManager contract. One important thing to note is that in order to call this function, the information on the latest block number until which the operator is required to serve the middleware needs to be passed.
+
+The following figure illustrates the above flow in which the operator calls on a `deregister(..)` function in a sample Registry contract.
+![Operator deregisteringn](./images/operator_deregister.png)
+
+### *Staker opting-in with a middleware*
+In order for any staker to participate in responding to the tasks from the middleware without running the middleware's off-chain software container themself, the staker can pursue delegation. The sequential flow for delegation process is as follows:
+1. The staker first has to delegate its stake to an EigenLayer operator who is opted-in to that middleware and that they trust (based on their own due diligence). The staker can accomplish this by calling the API `delegateTo(..)` in the DelegationManager contract, where the operator's address is passed as the argument.  
+2. Observe that with the staker delegating its stake to the operator, the weight associated with the operator's response to the task in the middleware it has opted-in increases. Therefore, the middleware must be notified about this. When the middleware is notified of this update, the call the `recordStakeUpdate` functionality to recognize the change in stake for that operator. On successful execution of this API call, the event `MiddlewareTimesAdded(..)` is emitted.
+
+The following figure illustrates the above flow: 
+![Staker opting-in](./images/staker_opting.png)
+
+
+### *Staker withdraws*
```

Comment:
if this document is intended for AVS developers, how much detail do they actually need in this section? seems good for them to have a gist of this, but IMO they don't need all the details.

Reply:
I think Soubhik felt it would be useful

Reply:
I think a summary from developer perspective + linking to detailed docs would be more appropriate. I'm OK with this as-is though -- it's not too wordy + I like the diagram